src/badguy/fish_chasing.cpp

```diff
@@ -73,16 +81,24 @@ FishChasing::active_update(float dt_sec) {
   // Behavior - chase the nearest player.
   auto player = get_nearest_player();
   if (!player) return;
-  Vector p1 = m_col.m_bbox.get_middle();
-  Vector p2 = player->get_bbox().get_middle();
-  Vector dist = (p2 - p1);
+  const Vector p1 = m_col.m_bbox.get_middle();
+  const Vector p2 = player->get_bbox().get_middle();
+  const Vector dist = (p2 - p1);
+  const bool is_player_in_water = player->is_swimming() || player->is_swimboosting() || player->is_water_jumping();
+  const bool is_facing_player = (m_dir == Direction::LEFT && dist.x <= 0.0f) || (m_dir == Direction::RIGHT && dist.x >= 0.0f );
+
+  const CollisionSystem::RaycastResult player_collision_check { Sector::get().get_first_line_intersection(p1, p2, false, get_collision_object()) };
+
+  const auto* obj = std::get_if<CollisionObject*>(&player_collision_check.hit);
+
+  const bool is_direct_view_to_player_possible { obj && *obj == player->get_collision_object() };
```

Comment:
This variable name is a bit too long. Maybe it can be renamed to something like `can_see_player`?

Additionally, `=` should be used.